### ISSUE TITLE
parseISO doesn't support time-only ISO string

### ISSUE THREAD

**Issue by scscgit** [date-fns/date-fns#2160]
Unlike [formatISO](https://date-fns.org/docs/formatISO), the [parseISO](https://date-fns.org/docs/parseISO) doesn't support time-only representation, and there is nothing related to this limitation mentioned in the docs (along with a lack of any example of a best practice regarding architecture that requires time-only representation, preferably with a proper support of time offsets or even timezones (to allow DST), e.g. when the UI uses some [time picker](https://vuetifyjs.com/en/components/time-pickers/) with a value in ISO 8601 format).
```ts
parseISO(
  formatISO(
    new Date(),
    { representation: 'time' }
  )
)
```
This returns a `Date` of `Invalid Date`. Are we supposed to always concatenate the time with an arbitrary date before using the resulting `Date` to [format](https://date-fns.org/docs/format) it, or should we always hard-code the parsing logic without respecting timezones? The only other helper function with a relevant name I found was [getTime](https://date-fns.org/docs/getTime), which doesn't help either, and I haven't found any similar issues or feature requests, so I hope we can come up with some consensus for new users.

**Comment by scscgit** [comment]
@ristomatti doesn't this lack the option of timezones?

**Comment by ristomatti** [comment]
> @ristomatti doesn't this lack the option of timezones?

I'm really not the right person to answer that, but isn't timezone support out of scope of this library but handled with [fns-tz](https://github.com/marnusw/date-fns-tz) or similar?

Also it seems I likely misunderstood your original issue, I believed what I suggested was what you were after. So I assume the issue is still valid then.

**Comment by scscgit** [comment]
Yeah there are both the matters of offsets and timezones, and the tz library doesn't help either. At best one can try to implement their own framework to parse something like `11:33[America/New_York]` and hard-code any offset translations, while worrying about accidentally failing to account for local timezone of user's browser, in which case the business logic of your application can end up creating dangerous hazards, especially if you aren't expert at JS. If the library doesn't cover common use-cases at all, we have no way of knowing what's the correct usage, especially when persisting the time to database. (We can't reasonably persist each user's own local time, can we now?)

**Comment by fturmel** [comment]
@scscgit There are timezone-related tokens in the `parse` function. https://date-fns.org/v2.25.0/docs/parse

Maybe something like this will work for you?

```js
console.log(parse("19:15:20-04:00", "HH:mm:ssXXX", new Date()));
// Wed Nov 03 2021 19:15:20 GMT-0400 (Eastern Daylight Time)
```

**Comment by fturmel** [comment]
Some supplemental details:

The first thing the `parseISO` function does is try to parse the date (year/month/date) segment of the string, so the current implementation will not accept time without a date as context. `parse` on the other hand, takes a date instance as a third argument to be able to complete missing information.

https://github.com/date-fns/date-fns/blob/f8fb243c55853f90d04efd8933fc291a24402b27/src/parseISO/index.js#L103-L111

I think all the examples in the ISO 8601 [wikipedia page](https://en.wikipedia.org/wiki/ISO_8601) should work with `parseISO` except for "date without year". There is time zone support in there, but you will not get a valid result from the function without providing date information as well.

<img width="292" alt="Screen Shot 2021-11-03 at 2 20 46 PM" src="https://user-images.githubusercontent.com/4009209/140169579-71a512f1-cbe2-4c01-a980-c2a4855c677f.png">

Hope this helps!
